src/documents/components/FlatFile/FlatFile.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import {Button, Popup} from 'semantic-ui-react';
+import {FlatfileButton} from '@flatfile/react';
+const {parse} = require('json2csv');
+/**
+ * FlatFile component for column mapping before document created
+ */
+
+const FlatFile = ({flatfileSettings, createVersion, history}) => {
+  if (flatfileSettings.length > 0 && !flatfileSettings.startsWith('ERROR: ')) {
+    return (
+      <Button
+        as={FlatfileButton}
+        labelPosition="right"
+        icon="lightbulb"
+        content="Try New Upload"
+        color="violet"
+        licenseKey="e118c06a-55ed-47be-90e4-26feb38c3934"
+        customer={{
+          companyId: 'ABC-123',
+          companyName: 'ABC Corp.',
+          email: 'john@abc123.com',
+          name: 'John Smith',
+          userId: '12345',
+        }}
+        settings={JSON.parse(flatfileSettings)}
+        onData={async results => {
+          console.log(results.allData);
+          const processedData = results.allData.map(row => {
+            const keys = Object.keys(row);
+            var formatedRow = {};
+            keys.map(key => {
+              if (key.startsWith('$')) {
+                Object.keys(row[key]).map(k => {
+                  formatedRow[k] = row[key][k];
+                  return true;
+                });
+              } else {
+                formatedRow[key] = row[key];
+              }
+              return true;
+            });
+            return formatedRow;
+          });
+          console.log(processedData);
+          console.log(parse(processedData));
+          var file = new File([parse(processedData)], 'foo.csv', {
```

Comment:
You can grab the original file's name from https://flatfile.com/developers/sdk/classes/flatfileresults/#filename and use it here